### PR TITLE
Problem: missing codecov.yml

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#
```
Comment (yihuang) [inline codecov.yml:1]: is the filename should be `.codecov.yml`?

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      default:
+        threshold: 1% # allow this much decrease on project
+      app:
+        target: 70%
+        flags:
+          - app
+      modules:
+        target: 70%
+        flags:
+          - modules
```
Comment (yihuang) [inline codecov.yml:23]: is it the module directory? should it be `x`?

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      client:
+        flags:
+          - client
```
Comment (yihuang) [inline codecov.yml:26]: there's no go code in client, maybe `cmd`?

File: codecov.yml (hunk 1 above)
Comment (thomas-nguy) [reply codecov.yml]: both should work but yes, let me rename it for consistency with the other config files

File: codecov.yml (hunk 2 above)
Comment (thomas-nguy) [reply codecov.yml]: what do you mean? the path for the flag is defined below

File: .codecov.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        target: 70%
+        flags:
+          - app
+      modules:
+        target: 70%
+        flags:
+          - modules
+      client:
+        flags:
+          - client
```
Comment (yihuang) [inline .codecov.yml:26]: it's `clients` in the `flags` section, is that expected?

Comment (thomas-nguy) [reply .codecov.yml]: thanks, yes this is a typo

**Review by yihuang — APPROVED** [review]

**Pull request merged**